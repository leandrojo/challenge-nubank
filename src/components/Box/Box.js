import styled from 'styled-components';

export function styleProps(props) {
  return props.css;
}

export const Flex = styled.div`
  display: flex;
  flex-direction: row;
`;

const Box = styled.div`
  display: flex;
  flex: 1;

  ${styleProps};
`;

Box.defaultProps = {
  css: {},
};

export default Box;
