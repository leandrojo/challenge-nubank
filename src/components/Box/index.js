export { default, styleProps, Flex } from './Box';
