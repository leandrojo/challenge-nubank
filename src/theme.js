const fontFamily = 'Rubik, sans-serif';

const theme = {
  backgroundColor: '#f5f5f5',
  colors: {
    primary: '#7f39b7',
    grayExtraDark: '#555',
    grayDark: '#767676',
    gray: '#999',
    grayLight: '#ccc',
    grayExtraLight: '#FAFAFA',
  },
  fontFamily,
  fontSize: '1em',
  typography: {
    heading: {
      color: '#252226',
    },
    paragraph: {
      color: '#716d73',
    },
  },
};

theme.components = {
  input: {
    css: {
      backgroundColor: 'transparent',
      fontFamily,
      fontSize: '1.4em',
      fontWeight: 300,
    },
  },
  pane: {
    backgroundColor: '#fff',
  },
  range: {
    emptyColor: theme.colors.grayLight,
  },
};

export default theme;
