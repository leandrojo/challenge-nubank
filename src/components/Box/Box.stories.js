/* eslint-disable import/no-extraneous-dependencies */

import React from 'react';

import Box from './Box';

export default {
  title: 'components/Box',
};

export const custom = () => (
  <Box
    css={`
      background: #fff;
      border: 1px dotted blue;
      display: block;
      padding: 20px;
    `}
  >
    <p>Box is open styled component custom. Here is custom background, border and padding.</p>
    <p>Use only as wrapper.</p>
  </Box>
);

custom.story = {
  name: 'custom',
};
