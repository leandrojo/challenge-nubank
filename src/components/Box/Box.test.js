/* eslint-env jest */

import React from 'react';
import { ThemeProvider } from 'styled-components';
import renderer from 'react-test-renderer';

import theme from 'theme';

import Box from './Box';

describe('Box', () => {
  it('with text', () => {
    const Component = () => {
      const content = 'Test Message';

      return (
        <ThemeProvider theme={theme}>
          <Box>{content}</Box>
        </ThemeProvider>
      );
    };

    const expected = renderer.create(<Component />).toJSON();

    expect(expected).toMatchSnapshot();
  });
});
