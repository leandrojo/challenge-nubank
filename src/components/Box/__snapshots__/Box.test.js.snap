// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box with text 1`] = `
<div
  className="sc-bwzfXH kaKVTG"
>
  Test Message
</div>
`;
